tests/Providers/Stripe/Customer/ServiceTest.php

```diff
@@ -89,4 +88,24 @@ public function testItCanCreateCustomerWithCardToken()
 
         $this->assertInstanceOf(Customer::class, $customer);
     }
+
+    /**
+     * Test card can be deleted.
+     */
+    public function testCanDeleteCard()
+    {
+        /** @var StripeCustomer $response */
+        $response = StripeCustomer::create([
+            'email' => 'software@pod-point.com',
+            'description' => 'test',
+        ]);
+
+        $card = $this->service->customers()->addCard($response->id, 'pm_card_visa');
```

Comment:
Would be nice to have a more fluent service class maybe.
```php
$this->service->customer($customer->id)->addCard('pm_card_visa');
```
or
```php
$this->service->customers()->find($customer->id)->addCard('pm_card_visa');
```
or
```php
$this->service->customers()->forCustomer($customer->id)->addCard('pm_card_visa');
```

Reply:
I disagree we had a whole discussion about that already and we agreed to do it that way because of the way the Stripe lib is built. Finding a customer before adding the card will add more loading everything is done at once.

Reply:
I get your point about avoiding multiple API calls but this is different from having this using fluent functions.

Reply:
As @LeeBrooks3 explained:

> we opted for all the "doing" (creating/retrieving/updating/deleting) to be within the services so that the models can remain as plain old php objects.